### ISSUE TITLE
`expr "$x" : '^$y"` is not reported as non-posix

### ISSUE THREAD

**Issue by cD1rtX3** [koalaman/shellcheck#3359]
#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
expr "X$1" : '^Xspecial'
```

#### Here's what shellcheck currently says:
Nothing.

#### Here's what I wanted or expected to see:
Something like “`expr` already only matches at the start of the string, so the `^` anchor is unspecified in POSIX sh”.

**Comment by rexingQ** [comment]
https://github.com/koalaman/shellcheck/issues/3359#issue-3629248777

**Comment by rexingQ** [comment]
UQBxWc5xdgbVZ4hgKxNX68LXePKyCKhtV03KdpMWmsvVmMnP
